cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -863,3 +863,133 @@ void Refactor::RenameType(mgp_list *args, mgp_graph *memgraph_graph, mgp_result
+  const auto arguments = mgp::List(args);
+  auto graph = mgp::Graph(memgraph_graph);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto nodes = arguments[0].ValueList();
+    if (nodes.Empty()) {
+      throw mgp::ValueException(std::string(kMergeNodesEmptyListError).c_str());
+    }
+
+    // Verify all elements are nodes
+    for (const auto &node_value : nodes) {
+      if (!node_value.IsNode()) {
+        throw mgp::ValueException(std::string(kMergeNodesInvalidTypeError).c_str());
+      }
+    }
+
+    const auto config = arguments[1].ValueMap();
+    bool mergeRels = false;
+    if (config.KeyExists(kMergeNodesRelationshipsStrategy)) {
+      if (!config.At(kMergeNodesRelationshipsStrategy).IsBool()) {
+        throw mgp::ValueException(std::string(kMergeRelationshipsInvalidValueError).c_str());
+      }
+
+      mergeRels = config.At(kMergeNodesRelationshipsStrategy).ValueBool();
+    }
+
+    // Get properties strategy from config
+    std::string prop_strategy = std::string(kMergeNodesPropertiesCombine);
+    if (config.KeyExists(kMergeNodesPropertiesStrategy)) {
+      prop_strategy = config.At(kMergeNodesPropertiesStrategy).ValueString();
+    }
+    if (config.KeyExists(kMergeNodesPropertiesStrategyAlternative)) {
+      prop_strategy = config.At(kMergeNodesPropertiesStrategyAlternative).ValueString();
+    }
+
+    const auto mergeNodesPropertiesCombineString = std::string(kMergeNodesPropertiesCombine);
+    const auto mergeNodesPropertiesDiscardString = std::string(kMergeNodesPropertiesDiscard);
+    const auto mergeNodesPropertiesOverrideString = std::string(kMergeNodesPropertiesOverride);
+    const auto mergeNodesPropertiesOverwriteString = std::string(kMergeNodesPropertiesOverwrite);
+
+    // Convert to lowercase for case-insensitive comparison
+    std::transform(prop_strategy.begin(), prop_strategy.end(), prop_strategy.begin(),
+                   [](unsigned char c) { return std::tolower(c); });
+
+    // Validate property strategy
+    if (prop_strategy != mergeNodesPropertiesCombineString && prop_strategy != mergeNodesPropertiesDiscardString &&
+        prop_strategy != mergeNodesPropertiesOverrideString && prop_strategy != mergeNodesPropertiesOverwriteString) {
+      throw mgp::ValueException(std::string(kMergeNodesInvalidPropertyStrategyError).c_str());
+    }
+
+    // Get the first node as the target node
+    auto target_node = nodes[0].ValueNode();
+
+    // Process remaining nodes
+    for (size_t i = 1; i < nodes.Size(); ++i) {
+      auto source_node = nodes[i].ValueNode();
+
+      // Handle properties based on strategy
+      if (prop_strategy == mergeNodesPropertiesCombineString) {
+        // Combine properties from both nodes
+        auto source_props = source_node.Properties();
+        for (const auto &[key, value] : source_props) {
+          if (!source_props.contains(key)) {
+            // nothing to combine
+            continue;
+          }
```

Comment:
I think this is wrong. You are iterating over source_props key-value pairs and checking source_props contains key